### ISSUE TITLE
Support STEP import in meshing tool

### ISSUE THREAD

**Issue by ddavidebor** [nextfoam/baram#45]
**Is your feature request related to a problem? Please describe.**
it's currently tedious for complex mechanical assemblies to export all the STLs. Please support STEP import in BaramMesh

**Describe the solution you'd like**
When clicking on "import" step files are supported
This can be achieved using OpenCASCADE technology (OCCT) CAD kernel

**Comment by jiban** [comment]
Thank you for your interest in BARAM.
Could you please tell me your current process to generate STL files for BARAM?
Are you using [Salome](https://www.salome-platform.org/) for it?
As you may know, you can select all the STL files at once in BARAM import menu.

**Comment by ddavidebor** [comment]
Hi,

Currently I'm exporting directly from my CAD software, as Salome is quite difficult to install and use

**Comment by jiban** [comment]
Sounds good because I found that Salome can export its geometries one by one only.
May I ask you which CAD software you are using and if it supports exporting geometries in group?

**Comment by ddavidebor** [comment]
I use Solidworks and Fusion360. All CAD software can export a step file with assemblies and sub-assemblies. 
And because STEP is a CAD format and not a mesh, you can easily select individual faces or bodies.

**Comment by jiban** [comment]
I've looked into [SolidWorks manual](https://help.solidworks.com/2021/english/SolidWorks/sldworks/hidd_stl.htm) and studied how I could implement the import function.
I found that SolidWorks exported the assemblies as an STL file, and this file would be imported into BaramMesh.
If STEP import function is implemented in BaramMesh, it would be almost same to that in SolidWorks.
Maybe I'm missing some part that is in the export part of SolidWorks.
Could you tell me what it is?

**Comment by ddavidebor** [comment]
I do not remember with Solidworks (I can check) but with fusion and most other CAD software you can only export one body at a time and the origin is lost. Also, you cannot export faces.

**Comment by jiban** [comment]
Yes. It definitely is tedious if we cannot export assemblies at once.
Please confirm that the way SolidWorks is doing is what you meant, and there is no problem if you use SolidWorks.

**Comment by jiban** [comment]
Dear @ddavidebor, I could not find proper STEP file for CFD, a STEP file that has all the boundaries and clean.
Could you send me a STEP file that, you think, could be a standard for the import function?
